expression/constant.go

```diff
@@ -198,7 +286,13 @@ func (c *Constant) EvalString(ctx sessionctx.Context, _ chunk.Row) (string, bool
 
 // EvalDecimal returns decimal representation of Constant.
 func (c *Constant) EvalDecimal(ctx sessionctx.Context, _ chunk.Row) (*types.MyDecimal, bool, error) {
-	if c.DeferredExpr != nil {
+	if c.ParamMarker != nil {
+		dt := c.ParamMarker.Value()
+		if dt.IsNull() {
+			return nil, true, nil
+		}
+		c.Value.SetValue(dt.GetValue())
```

Comment:
No need convert `dt` to `decimal` first?

Reply:
current code just follow old logic https://github.com/pingcap/tidb/blob/751df9789ed085f08d900b59a5630d1d176885e2/expression/constant.go#L309 will do a convert, maybe we can optimize this in later